restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +86,30 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  public static void getAbsentFieldsDefaultValues(RecordDataSchema dataSchema, DataMap dataMap)
+  {
+    for (RecordDataSchema.Field field : dataSchema.getFields())
+    {
+      if (!dataMap.containsKey(field.getName()))
+      {
+        if (field.getDefault() != null)
+        {
+          dataMap.put(field.getName(), field.getDefault());
+        }
+        else
+        {
+          DataSchema fieldSchema = field.getType();
```

Comment:
test this for typerefs

Reply:
Got an exception:
```
java.lang.ClassCastException: com.linkedin.data.schema.TyperefDataSchema cannot be cast to com.linkedin.data.schema.RecordDataSchema
```

Reply:
yeah, you might want to use dereferenced schema for typerefs

Reply:
Resolved this, Aman!